src/FuseGPUThreadLoops.cpp

```diff
@@ -352,6 +352,24 @@ class ExtractSharedAllocations : public IRMutator {
         stmt = op->body;
     }
 
+    void visit(const AddressOf *op) {
+        if (shared.count(op->name)) {
+            Expr index = mutate(op->index);
+            shared[op->name].max = barrier_stage;
+            if (device_api == DeviceAPI::OpenGLCompute) {
+                expr = AddressOf::make(op->type, shared_mem_name + "_" + op->name,
```

Comment:
I think the first arg is wrong here - it's not a pointer type. That said, I think the first arg should go away anyway.